### PR TITLE
Allow passing multiple metadata key-value pairs

### LINKED ISSUE #548: Extend AnalyticsPingManager to accept multiple metadata key-value pairs

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #548.

I'll do actual testing in the meantime.

### CODE REVIEW

**Comment by chanseokoh** [comment]
Things are working fine.

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java [hunk 1]
```diff
@@ -158,35 +164,59 @@ private void registerOptInStatus(boolean optedIn) {
...
-   * String, String, String, Shell)} and pass the {@Shell} of the currently open modal dialog.
+   * are calling this method inside another modal dialog, consider using {@link #sendPingOnShell(
+   * Shell, String, ImmutableMap)} and pass the {@Shell} of the currently open modal dialog.
    * (Otherwise, the opt-in dialog won't be able to get input until the workbench can get input.)
    *
    * Safe to call from non-UI contexts.
    */
-  public void sendPing(String eventName, String metadataKey, String metadataValue) {
-    sendPing(eventName, metadataKey, metadataValue, null);
+  public void sendPing(String eventName, ImmutableMap<String, String> metadata) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: Requiring this to be `ImmutableMap` seems overly restrictive?

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: Probably. I should just copy the map.

File: plugins/com.google.cloud.tools.eclipse.usagetracker.test/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManagerTest.java [hunk 2]
```diff
@@ -58,29 +62,76 @@ public void setUp() {
...
+    try {
+      new PingEvent(null, EMPTY_MAP, null);
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertEquals("eventName null or empty", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testPingEventConstructor_emtpyEventName() {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.usagetracker.test/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManagerTest.java]: sp: emptyEvent

**Pull request merged**